src/python/TaskWorker/MasterWorker.py

```diff
@@ -400,6 +487,8 @@ def algorithm(self):
         self.restartQueuedTasks()
         self.logger.debug("Master Worker Starting Main Cycle.")
         while not self.STOP:
+            selection_limit = 10 #TBD what should be the value
```

Comment:
since this is the only hardcoded number and indeed we do not know what to set, maybe better to add it to the configuration ? I would add an ad-hoc section there, where we can also have parameters for the external scheduler